netplugin/plugin/netplugin.go

```diff
@@ -303,27 +304,57 @@ func (p *NetPlugin) Reinit(cfg Config) {
 }
 
 //InitGlobalSettings initializes cluster-wide settings (e.g. fwd-mode)
-func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) {
+func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) error {
+
+	/*
+		Query global settings from state store
+		1. if forward mode or private net is empty, retry 1 sec
+		2. if forward mode doesn't match, return error
+		3. if stored private net is wrong, return error
+	*/
 
 	gCfg := mastercfg.GlobConfig{}
 	gCfg.StateDriver = stateDriver
-	err := gCfg.Read("")
-	if err != nil {
-		logrus.Errorf("Error reading forwarding mode from cluster store")
-		inst.FwdMode = "bridge"
-		inst.HostPvtNW = defaultPvtSubnet
-		return
+
+	// wait until able to get fwd mode and private subnet
+	for {
+		err := gCfg.Read("")
+		if err != nil {
+			logrus.Errorf("Error reading global settings from cluster store")
+		}
+		if gCfg.FwdMode == "" || gCfg.PvtSubnet == "" {
+			if gCfg.FwdMode == "" {
+				logrus.Errorf("Error reading forwarding mode from cluster store")
+			} else {
+				logrus.Errorf("Error reading private subnet cluster store")
+			}
+
+		} else {
+			logrus.Infof("Get global forwarding mode: %v", gCfg.FwdMode)
+			logrus.Infof("Get global private subnet: %v", gCfg.PvtSubnet)
+			break
+		}
+		logrus.Infof("Sleep 1 second and retry pulling global settings")
+		time.Sleep(1 * time.Second)
+	}
+
+	if inst.FwdMode == "" {
+		inst.FwdMode = gCfg.FwdMode
+	} else {
+		if inst.FwdMode != gCfg.FwdMode {
```

Comment:
this check should be in netmaster/netctl.
netplugin being a worker node, it should accept the value from the netmaster

Reply:
technically local fwd mode will always be empty or the one get from kv store with this code, it doesn't hurt if anything wired happens here

Reply:
currently `inst.FwdMode` is `"" ` because during init there's no default value set